snforge_std/src/cheatcodes.cairo

```diff
@@ -22,83 +28,116 @@ fn test_address() -> ContractAddress {
+    validate_cheat_span(@span);
+
     let mut inputs = array![];
     target.serialize(ref inputs);
+    span.serialize(ref inputs);
     inputs.append(block_number.into());
     cheatcode::<'start_roll'>(inputs.span());
 }
 
+fn start_roll(target: CheatTarget, block_number: u64) {
+    roll(target, block_number, CheatSpan::Indefinite);
+}
+
 fn stop_roll(target: CheatTarget) {
     let mut inputs = array![];
     target.serialize(ref inputs);
     cheatcode::<'stop_roll'>(inputs.span());
 }
 
-fn start_prank(target: CheatTarget, caller_address: ContractAddress) {
+fn prank(target: CheatTarget, caller_address: ContractAddress, span: CheatSpan) {
+    validate_cheat_span(@span);
+
     let mut inputs = array![];
     target.serialize(ref inputs);
+    span.serialize(ref inputs);
     inputs.append(caller_address.into());
     cheatcode::<'start_prank'>(inputs.span());
 }
 
+fn start_prank(target: CheatTarget, caller_address: ContractAddress) {
+    prank(target, caller_address, CheatSpan::Indefinite);
+}
+
 fn stop_prank(target: CheatTarget) {
     let mut inputs = array![];
     target.serialize(ref inputs);
     cheatcode::<'stop_prank'>(inputs.span());
 }
 
-fn replace_bytecode(contract: ContractAddress, new_class: ClassHash) {
-    cheatcode::<'replace_bytecode'>(array![contract.into(), new_class.into()].span());
-}
+fn warp(target: CheatTarget, block_timestamp: u64, span: CheatSpan) {
+    validate_cheat_span(@span);
 
-fn start_warp(target: CheatTarget, block_number: u64) {
     let mut inputs = array![];
     target.serialize(ref inputs);
-    inputs.append(block_number.into());
+    span.serialize(ref inputs);
+    inputs.append(block_timestamp.into());
     cheatcode::<'start_warp'>(inputs.span());
 }
 
+fn start_warp(target: CheatTarget, block_timestamp: u64) {
+    warp(target, block_timestamp, CheatSpan::Indefinite);
+}
+
 fn stop_warp(target: CheatTarget) {
     let mut inputs = array![];
     target.serialize(ref inputs);
     cheatcode::<'stop_warp'>(inputs.span());
 }
 
-fn start_elect(target: CheatTarget, sequencer_address: ContractAddress) {
+fn elect(target: CheatTarget, sequencer_address: ContractAddress, span: CheatSpan) {
+    validate_cheat_span(@span);
+
     let mut inputs = array![];
     target.serialize(ref inputs);
+    span.serialize(ref inputs);
     inputs.append(sequencer_address.into());
     cheatcode::<'start_elect'>(inputs.span());
 }
 
+fn start_elect(target: CheatTarget, sequencer_address: ContractAddress) {
+    elect(target, sequencer_address, CheatSpan::Indefinite);
+}
+
 fn stop_elect(target: CheatTarget) {
     let mut inputs = array![];
     target.serialize(ref inputs);
     cheatcode::<'stop_elect'>(inputs.span());
 }
 
-fn start_mock_call<T, impl TSerde: core::serde::Serde<T>, impl TDestruct: Destruct<T>>(
-    contract_address: ContractAddress, function_selector: felt252, ret_data: T
+fn mock_calls<T, impl TSerde: core::serde::Serde<T>, impl TDestruct: Destruct<T>>(
```

Comment:
The interface of this is different than other cheatcodes. It hides the `CheatSpan` as the cheat works for `amount` calls to the mocked function, unlike other cheats working during the span of N calls.

Reply:
Imo it's fine since using the same enum for with different semantics can be misleading